xwiki-platform-core/xwiki-platform-like/xwiki-platform-like-api/src/main/resources/templates/likers.vm

```diff
@@ -17,106 +17,118 @@
+    #set ($grayed = $xcontext.userReference == $userReference.reference)
+    #set ($userDoc = $xwiki.getDocument($userReference.reference))
+    #set ($userProperties = $services.user.getProperties($userReference))
+    #set ($userObject = $user.getObject('XWiki.XWikiUsers'))
+    #if ($services.security.authorization.hasAccess('view', $userDoc.documentReference))
+      #set ($row = {
+        'grayed': $grayed,
+        'doc_fullName': $userDoc.fullName,
+        'doc_wiki': $userDoc.wiki,
+        'doc_url': $userDoc.getURL(),
+        'doc_viewable': true,
+        'name': "#displayUserAliasWithAvatar($userDoc.documentReference $disabled)",
+        'first_name': $userProperties.firstName,
+        'last_name': $userProperties.lastName
+      })
     #else
-      #set ($totalRows = $likeRecords.size())
+      #set ($row = {
+        'doc_viewable': false,
+        'doc_fullName': 'obfuscated'
+      })
     #end
-    
-    #set ($map = {
-        "totalrows": $totalRows,
-        "returnedrows": $likeRecords.size(),
-        "offset": $newOffset,
-        "reqNo": $numbertool.toNumber($request.reqNo),
-        "rows": $userRows
-    })
-    #livetable_filterObfuscated($map)
-    $jsontool.serialize($map)
-#else
-    <h1>$escapetool.xml($services.localization.render('like.likers.title', [$doc.plainTitle]))</h1>
-    #set($columns = ["name", "first_name", "last_name"])
-    #set($columnsProperties = {
-        "name" : { "type" : "text", "sortable": false, "filterable": false, "html": true },
-        "first_name" : { "type" : "text", "sortable": false, "filterable": false},
-        "last_name" : { "type" : "text", "sortable": false, "filterable": false}
-    })
-    #set ($queryParams = {
-        "livetable": "true",
-        "xpage": "likers",
-        "outputSyntax": "plain"
-    })
-    ## We rely on the same column name than the Users administration, so we use same translation prefix for now.
-    #set($options = {
-        'url': $doc.getURL('get', $escapetool.url($queryParams)),
-        'outputOnlyHtml': true,
-        'translationPrefix' : "xe.admin.users."
-    })
+    #set ($discard = $userRows.add($row))
+  #end
+  ## ===
+  ## JSON
+  ## ===
+  #set ($newOffset = $offset + 1)
+  #set ($optLikesNumber = $services.like.getLikes($documentReference))
+  #if ($optLikesNumber.isPresent())
+    #set ($totalRows = $optLikesNumber.get())
+  #else
+    #set ($totalRows = $likeRecords.size())
+  #end
+  
+  #set ($map = {
+    "totalrows": $totalRows,
+    "returnedrows": $likeRecords.size(),
+    "offset": $newOffset,
+    "reqNo": $numbertool.toNumber($request.reqNo),
+    "rows": $userRows
+  })
+  #livetable_filterObfuscated($map)
+  #jsonResponse($map)
+#end
+{{/velocity}}{{velocity}}##A line break is inserted if the two macro are not closing and opening on the same line.
+  #if ($request.livetable == "true")
+    #break
+  #end
+  = $escapetool.xml($services.localization.render('like.likers.title', [$doc.plainTitle])) =
```

Comment:
This seems like the wrong escaping for XWiki syntax.

Reply:
Thanks